sql/parser/builtins.go

```diff
@@ -347,6 +348,80 @@ var builtins = map[string][]builtin{
 	"max": aggregateImpls(boolType, intType, floatType, stringType),
 	"min": aggregateImpls(boolType, intType, floatType, stringType),
 	"sum": aggregateImpls(intType, floatType),
+
+	// Timestamp/Date functions.
+
+	"age": {
+		builtin{
+			types:      typeList{timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: time.Now().Sub(args[0].(DTimestamp).Time)}, nil
+			},
+		},
+		builtin{
+			types:      typeList{timestampType, timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: args[0].(DTimestamp).Sub(args[1].(DTimestamp).Time)}, nil
+			},
+		},
+	},
+
+	"current_date": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyDate,
+			fn: func(args DTuple) (Datum, error) {
+				return DDate{Time: time.Now().Truncate(24 * time.Hour)}, nil
+			},
+		},
+	},
+
+	"clock_timestamp":   {nowBuiltin()},
+	"current_timestamp": {nowBuiltin()},
+	"localtimestamp":    {nowBuiltin()},
+	"now":               {nowBuiltin()},
+
+	"extract": {
+		builtin{
+			types:      typeList{stringType, timestampType},
+			returnType: DummyFloat,
+			fn: func(args DTuple) (Datum, error) {
+				t := args[1].(DTimestamp)
+				s := strings.ToLower(string(args[0].(DString)))
+				switch s {
+				case "year":
+					return DFloat(t.Year()), nil
```

Comment:
One thought is to convert the return type to int64 and change the semantics of second, millisecond, microsecond so that we can return a int64. I'll also add nanosecond.